mops/include/mops/sap.hpp

```diff
@@ -0,0 +1,60 @@
+#ifndef MOPS_OUTER_PRODUCT_SCATTER_ADD_HPP
+#define MOPS_OUTER_PRODUCT_SCATTER_ADD_HPP
+
+#include <cstddef>
+#include <cstdint>
+
+#include "mops/exports.h"
+#include "mops/tensor.hpp"
+
+namespace mops {
+    /// TODO
+    template<typename scalar_t>
+    void MOPS_EXPORT sparse_accumulation_of_products(
+        Tensor<scalar_t, 2> output,
+        Tensor<scalar_t, 2> tensor_a,
+        Tensor<scalar_t, 2> tensor_b,
+        Tensor<scalar_t, 1> tensor_c,
+        Tensor<int32_t, 1> p_a,
```

Comment:
what does `p` stands for here? Should we use `indices_a`/`indices_b`/... instead?

Reply:
I would call it tensor_p_a if we want to have a more representative name. P_A is also an option, and it would be consistent with Python. An issue I see with having different names is that then error messages will look confusing coming from C++ to a Python user